### PR TITLE
Neomake instead Syntastic in NeoVim

### PR DESCRIPTION
**Pull request by renatoagds**
**CHANGE LOG**

This PR uses [Neomake](https://github.com/neomake/neomake) instead Syntastic to lint the files, is a big improvement in performance, since Neomake lint the file asynchronous.

### CODE REVIEW

**Comment by sloanelybutsurely** [comment]
Thank you for this, @renatoagds. I'm going to look at this either today or tomorrow.

File: vimrc [hunk 1]
```diff
@@ -23,6 +23,33 @@ if has('nvim')
   let g:tern_request_timeout = 1
   let g:tern_show_signature_in_pum = 0
   set completeopt-=preview
+
+  " == neomake/neomake ==
+  let g:neomake_javascript_jshint_maker = {
```
Comment (sloanelybutsurely) [inline vimrc:28]: Is this doing anything if the only enabled maker is eslint?

**Review by sloanelybutsurely — CHANGES_REQUESTED** [review]
This is really exciting. Do you have time to work on getting this to work for flow as well? If not, I can take over when I get a chance. Would be happy to merge this once we know if that jshint line is required and we can use it for Flow type checking as well.

**Comment by sloanelybutsurely** [comment]
@renatoagds I tried this out and it was soooo much faster for both eslint and flow checking. I just couldn't wait so I merged 😄 

Thanks you for this contribution 🙏

File: vimrc (hunk 1 above)
Comment (renatoagds) [reply vimrc:28]: @zperrault `neomake` naming is pretty bad, but this one is used for `eslint` too :)

Comment (davidosomething) [reply vimrc:28]: that is not true, `g:neomake_javascript_eslint_maker` is used to override the pre-existing eslint maker

Comment (renatoagds) [reply vimrc:28]: @davidosomething you're right! mb, thanks to fix my comment 👍

**Pull request merged**